### PR TITLE
removed pinata package and added custom api call function

### LINKED ISSUE #1569: Technical issues
Things to look into

- Launch is too slow 
- Reload is slow 
- Typescript 
- Bundle size

### PR DESCRIPTION
**Pull request by nenadV91**
# Summary

Fixes #1569, #2449

This enable us to remove pinata package and calls the pinata api directly which reduces the build size

  # To Test

I'm not sure how this affiliate flow actually works so someone who is more familiar with this please test it in details. It should work the same as it is working on production.

### CODE REVIEW

File: src/custom/api/ipfs/index.ts [hunk 1]
```diff
@@ -1,8 +1,35 @@
-import pinataSDK, { PinataPinResponse } from '@pinata/sdk'
 import { PINATA_API_KEY, PINATA_SECRET_API_KEY } from 'constants/ipfs'
 
-const pinata = pinataSDK(PINATA_API_KEY, PINATA_SECRET_API_KEY)
+export async function pinJSONToIPFS(file: any): Promise<any> {
```
Comment (anxolin) [inline src/custom/api/ipfs/index.ts:3]: why did you change the return type? 

returning any doesn't look to good

File: src/custom/api/ipfs/index.ts [hunk 2]
```diff
@@ -1,8 +1,35 @@
...
+    const data = await response.json()
+
+    if (response.status !== 200) {
+      throw new Error(data.error.details || data.error)
+    }
+
+    return data
+  } catch (error) {
+    console.log('Failed to pin JSON to IPFS on pinata api', error)
+    return error
```
Comment (anxolin) [inline src/custom/api/ipfs/index.ts:33]: why catching for returning an error? 

the method returns a promise, should we just not catch?

File: src/custom/api/ipfs/index.ts [hunk 3]
```diff
@@ -1,8 +1,35 @@
-import pinataSDK, { PinataPinResponse } from '@pinata/sdk'
 import { PINATA_API_KEY, PINATA_SECRET_API_KEY } from 'constants/ipfs'
 
-const pinata = pinataSDK(PINATA_API_KEY, PINATA_SECRET_API_KEY)
+export async function pinJSONToIPFS(file: any): Promise<any> {
+  const pinataUrl = `https://api.pinata.cloud/pinning/pinJSONToIPFS`
```
Comment (matextrem) [inline src/custom/api/ipfs/index.ts:4]: Maybe this could be moved to a constants file?

Comment (anxolin) [reply src/custom/api/ipfs/index.ts:4]: at least the base URL

**Comment by elena-zh** [comment]
Hey @nenadV91 , I have verified the issue. Affiliate workflow works fine, and there are no endless `pinJSON...` requests in the Network tab. 

However, I noticed that a lot of /account/ BE requests are sent to the server not (up to 8 per minute). 
1 min

However, it is happening in Prod as well, so it does not related to this PR.

**Review by elena-zh — APPROVED** [review]

File: src/custom/api/ipfs/index.ts (hunk 1 above)
Comment (nenadV91) [reply src/custom/api/ipfs/index.ts:3]: My idea was if I import that type I also need to import the library but I will just add the type in our code

File: src/custom/api/ipfs/index.ts (hunk 2 above)
Comment (nenadV91) [reply src/custom/api/ipfs/index.ts:33]: I'm not sure, my idea was to catch it and do a console.log and then also return it, what do you think? This is what they are doing https://github.com/PinataCloud/Pinata-SDK/blob/fe4743db2a3e9f0d03240eccaa1719ebb0d4a962/src/util/errorResponse.js and then https://github.com/PinataCloud/Pinata-SDK/blob/fe4743db2a3e9f0d03240eccaa1719ebb0d4a962/src/commands/pinning/pinJSONToIPFS.js

File: src/custom/api/ipfs/index.ts [hunk 4]
```diff
@@ -1,8 +1,42 @@
...
-const pinata = pinataSDK(PINATA_API_KEY, PINATA_SECRET_API_KEY)
+type PinataPinResponse = {
+  IpfsHash: string
+  PinSize: number
+  Timestamp: string
+}
 
 export async function pinJSONToIPFS(file: any): Promise<PinataPinResponse> {
-  return pinata.pinJSONToIPFS(file, { pinataMetadata: { name: 'appData-affiliate' } })
+  const pinataUrl = `${PINATA_API_URL}/pinning/pinJSONToIPFS`
```
Comment (W3stside) [inline src/custom/api/ipfs/index.ts:10]: pinataUrl and headers should be outside the function

**Review by W3stside — APPROVED** [review]
minor comments

**Review by alfetopito — APPROVED** [review]
